File: rzops-web/src/lib/types/database_instance.ts
export interface DatabaseInstanceResponse {
  id: string;
  server_id: string | null;
  name: string;
  db_type: string;
  description: string | null;
  status: string;
  environment: string | null;
  offline_time: string | null;
  is_self_installed: boolean | null;
  importance: string | null;
  is_ops_managed: boolean | null;
  port: number | null;
  instance_name: string | null;
  created_at: string;
  updated_at: string;
}

export interface DatabaseInstanceListResponse {
  data: DatabaseInstanceResponse[];
  count: number;
}

export interface CreateDatabaseInstanceRequest {
  name: string;
  db_type: string;
  server_id?: string;
  description?: string;
  status?: string;
  environment?: string;
  offline_time?: string;
  is_self_installed?: boolean;
  importance?: string;
  is_ops_managed?: boolean;
  port?: number;
  instance_name?: string;
}

export interface UpdateDatabaseInstanceRequest extends Partial<CreateDatabaseInstanceRequest> {}

export interface ListDatabaseInstancesQuery {
  q?: string;
  status?: string;
  environment?: string;
  db_type?: string;
  server_id?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/database_instance.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;


/// DatabaseInstance entity — maps to cmdb_database_instance.
#[derive(Debug, Clone)]
pub struct DatabaseInstance {
    pub id: Uuid,
    pub server_id: Option<Uuid>,
    pub name: String,
    pub db_type: String,
    pub description: Option<String>,
    pub status: String,
    pub offline_time: Option<DateTime<Utc>>,
    pub is_self_installed: bool,
    pub importance: Option<String>,
    pub is_ops_managed: bool,
    pub backup_plan_id: Option<Uuid>,
    pub monitor_target_id: Option<Uuid>,
    pub port: Option<i32>,
    pub instance_name: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
